bluebrain/version-bumping/src/bumper.py

```diff
@@ -0,0 +1,301 @@
+        prefix = ", ".join(packages)
+        commit_message = f"{prefix}: new versions"
+        logger.debug(f"Committing with message {commit_message}")
+        repo.index.commit(commit_message, author=author, committer=author)
+        result = repo.remote("github").push(refspec=f"{COMMIT_BRANCH}:{COMMIT_BRANCH}")
+        if result[0].flags not in [result[0].FAST_FORWARD, result[0].NEW_HEAD]:
+            raise RuntimeError(f"Undesirable push result: {result[0].flags} - {result[0].summary}")
+
+    def repo(self):
+        """
+        Instantiate the repo if it isn't done yet, and add the github remote if necessary
+        """
+        # TODO: make sure "." is correct
+        if not self._repo:
+            self._repo = Repo(".")
+            try:
+                self._repo.remote("github")
+            except ValueError:
+                self._repo.create_remote(
+                    "github",
+                    "https://bbp-hpcteam:${GITHUB_API_KEY}@github.com/bluebrain/spack",
+                )
+        return self._repo
+
+    def file_pull_request(self, packages):
+        logger.info("===> Filing pull request")
+        url = "https://api.github.com/repos/bluebrain/spack"
+        session = requests.Session()
+        session.headers = {
+            "Authorization": f"token {os.environ['GITHUB_API_KEY']}",
+            "Accept": "application/vnd.github+json",
+            "X-GitHub-Api-Version": "2022-11-28",
+        }
+
+        logger.debug(f"Getting pull requests with head {COMMIT_BRANCH}")
+        pulls = session.get(f"{url}/pulls?base=develop&head={COMMIT_BRANCH}").json()
+        if not pulls:
+            logger.debug("PR doesn't exist yet - filing")
+            data = {
+                "title": f"{', '.join(packages)}: new releases",
+                "body": "Bumper found new releases, please check carefully and add new "
+                "dependencies, remove obsolete dependencies, ...",
+                "head": f"{COMMIT_BRANCH}",
+                "base": "develop",
+            }
+            response = session.post(f"{url}/pulls", json=data)
+
+            logger.debug("Pull request made")
+            logger.debug(response)
+            logger.debug(response.content)
+            response.raise_for_status()
+        else:
+            logger.debug("Pull request already exists: {pulls[0]['_links']['self']['href']}")
+
+    def process_packages(self):
+        """
+        Check packages for newer versions and update recipes if necessary
+
+        For all subscribed packages:
+          1. check what the latest version is
+          2. check where it lives
+          3. check what the latest version in source control is
+          4. if newer: add a version call to the package recipe
+        """
+
+        missing_versions = []
+        updated_packages = []
+        repo = self.repo()
+        for package in self.packages:
+            package_file = f"bluebrain/repo-bluebrain/packages/{package}/package.py"
```

Comment:
What about things that live in https://github.com/BlueBrain/spack/tree/develop/bluebrain/repo-patches/packages ? Hopefully that will be an increasing number, as things get upstreamed?

Reply:
I would hope that `repo-patches` will get *smaller* as things get upstreamed 😅 
Either way, it seems to be an extremely slow process, so I'm not going to worry too much about it just yet. Seems like a decent candidate for a future improvement though!